homeassistant/components/tag/__init__.py

```diff
@@ -0,0 +1,136 @@
+"""The Tag integration."""
+import copy
+import datetime as dt
+import logging
+import typing
+from typing import Optional, cast
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_ID, CONF_NAME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers import collection
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.storage import Store
+from homeassistant.loader import bind_hass
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+DEVICE_ID = "device_id"
+EVENT_TAG_SCANNED = "tag_scanned"
+LAST_SCANNED = "last_scanned"
+STORAGE_KEY = DOMAIN
+STORAGE_VERSION = 1
+TAG_ID = "tag_id"
+TAGS = "tags"
+
+CREATE_FIELDS = {
+    vol.Required(CONF_ID): cv.string,
+    vol.Optional(CONF_NAME): vol.All(str, vol.Length(min=1)),
+    vol.Optional("description"): cv.string,
+    vol.Optional(LAST_SCANNED): cv.datetime,
```

Comment:
Let's make this required.

Reply:
As discussed, if we let folks manually register these in a helper UI this may not be present for a create.